### PR TITLE
Topic/constraint python

### PR DESCRIPTION
**Pull request by pFernbach**
* add flag for the number of free variable in python
* add method computeEndEffectorVelocityCost to python API
* add method computeEndEffectorDistanceCost to python API
* add method computeEndEffectorConstraints to python API
* add method computeEndEffectorConstantWaypoints to python API

### CODE REVIEW

File: include/hpp/bezier-com-traj/solve_end_effector.hh [hunk 1]
```diff
@@ -309,66 +315,75 @@ void computeJerkCostFunctionDiscretized(int numPoints,const ProblemData& pData,d
...
+    MatrixXX A;
+    VectorX b;
     computeConstraintsMatrix(pData,wps_acc,wps_vel,acc_bounds,vel_bounds,A,b,wps_jerk,jerk_bounds);
-  //  std::cout<<"End eff A = "<<std::endl<<A<<std::endl;
- //   std::cout<<"End eff b = "<<std::endl<<b<<std::endl;
-    // compute cost function (discrete integral under the curve defined by 'path')
-    MatrixXX H_rrt,H;
-    VectorX g_rrt,g;
-    std::pair<MatrixXX,VectorX> Hg_smooth;
+    return std::make_pair<MatrixXX,VectorX>(A,b);
```
Comment (nim65s) [inline include/hpp/bezier-com-traj/solve_end_effector.hh]: Can you use `std::make_pair(A, b);` here ? The templated version doesn't work in c++11.

File: include/hpp/bezier-com-traj/solve_end_effector.hh [hunk 2]
```diff
@@ -309,66 +315,75 @@ void computeJerkCostFunctionDiscretized(int numPoints,const ProblemData& pData,d
...
     H = MatrixXX::Zero(DIM_VAR,DIM_VAR);
     g  = VectorX::Zero(DIM_VAR);
-    H = weightSmooth*(Hg_smooth.first) + weightDistance*H_rrt;
-    g = weightSmooth*(Hg_smooth.second) + weightDistance*g_rrt;
+    H = weightSmooth*(Hg_smooth.first) + weightDistance*Hg_rrt.first;
+    g = weightSmooth*(Hg_smooth.second) + weightDistance*Hg_rrt.second;
    // H = Hg_smooth.first;
   //  g = Hg_smooth.second;
+
+    return std::make_pair<MatrixXX,VectorX>(H,g);
```
Comment (nim65s) [inline include/hpp/bezier-com-traj/solve_end_effector.hh]: same here

File: include/hpp/bezier-com-traj/solve_end_effector.hh (hunk 1 above)
Comment (pFernbach) [reply include/hpp/bezier-com-traj/solve_end_effector.hh]: Done, thanks for noticing this.

File: include/hpp/bezier-com-traj/solve_end_effector.hh (hunk 2 above)
Comment (pFernbach) [reply include/hpp/bezier-com-traj/solve_end_effector.hh]: Done.

File: src/computeCOMTraj.cpp [hunk 3]
```diff
@@ -58,7 +58,7 @@ bezier_wp_t::t_point_t computeDiscretizedWwaypoints(const ProblemData& pData,dou
      Normalize(A,b);
      // add 1 for the slack variable :
      A.block(0,3,dimH,1) = VectorX::Ones(dimH);
-     return std::make_pair<MatrixXX,VectorX>(A,b);
+     return std::make_pair(A,b);
```
Comment (nim65s) [inline src/computeCOMTraj.cpp]: I was going to open a separate PR for this, thanks for having done it :D

**Pull request merged**